src/webapp/js/actions/index.js

```diff
@@ -380,3 +382,22 @@ export function fetchReports() {
     payload: request,
   };
 }
+
+export function fetchAutomatedReports() {
+  const url = '/api/job';
+  const request = apiClient.get(url);
+
+  return {
+    type: FETCH_AUTOMATED_REPORTS,
+    payload: request,
+  };
+}
+
+export function updateAutomatedReports(reportSettings, callback) {
+  const url = 'api/job/update';
+  const request = apiClient.put(url, reportSettings).then(() => callback());
```

Comment:
You need to save the request first to the const and then use the callback (in separate lines), like in the example I added before.
After you call **then** you will not get the promise